src/zimdump.cpp

```diff
@@ -325,7 +322,7 @@ void ZimDumper::dumpFiles(const std::string& directory, bool symlinkdump, std::f
     }
 
     std::stringstream ss;
-    ss << dir << SEPARATOR << filename;
+    ss << dir << filename;
```

Comment:
Now you can simply output `path` here, which means that the need for `ss` above and `relative_path` below has disappeared (or, maybe, `relative_path` can be preserved but simply initialized with `path`, since the name `relative_path` is more appropriate in that code).

Note, that `SEPARATOR` is different between Windows and POSIX, so eliminating it here (and relying on the '/' present in the original path) may not be that insignificant. However, this code contains other possibilities for the POSIX-style path separators to be passed into Windows API calls, so that issue, must be addressed, separately.

Reply:
`filename` is a shortened version of the actual filename in `path`. Using `path` directly can cause `createdir()` to fail if the name is too long. eg, this failed:
```
-> dir: A/wiki.openzim.org/w/, 
-> filename: load.phpdebug=false&lang=en&modules=ext.eventLogging.subscriber%7Cext.visualEditor.desktopArticleTarget.init%7Cext.visualEditor.supportCheck%2CtargetLoader%2CtempWikitextEditorWidget%2Ctrack%2Cve%7Cjquery.accessKeyLabel%2CcheckboxShiftClick%2Cclient%2C~1, 
-> path: A/wiki.openzim.org/w/load.php?debug=false&lang=en&modules=ext.eventLogging.subscriber%7Cext.visualEditor.desktopArticleTarget.init%7Cext.visualEditor.supportCheck%2CtargetLoader%2CtempWikitextEditorWidget%2Ctrack%2Cve%7Cjquery.accessKeyLabel%2CcheckboxShiftClick%2Cclient%2Ccookie%2CgetAttrs%2ChighlightText%2Cmw-jump%2Csuggestions%2CtabIndex%2Ctablesorter%2CtextSelection%2Cthrottle-debounce%7Cmediawiki.RegExp%2CString%2CTitle%2CUri%2Capi%2Ccldr%2Clanguage%2Cnotify%2CsearchSuggest%2Cstorage%2Ctemplate%2Cuser%2Cutil%7Cmediawiki.api.options%2Cuser%7Cmediawiki.editfont.styles%7Cmediawiki.language.data%2Cinit%2Cmonths%7Cmediawiki.libs.pluralruleparser%7Cmediawiki.page.ready%2Cstartup%7Cmediawiki.template.regexp%7Cskins.vector.js%7Cuser.defaults&skin=vector&version=1fq761s
```

Yes, `SEPARATOR` should not be removed. Thank you @veloman-yunkan for pointing it out.